plugins/processors/encoding/encoding.go

```diff
@@ -0,0 +1,210 @@
+	Encoding         string `toml:"encoding"`
+
+	Log       telegraf.Logger `toml:"-"`
+	operation func(string, string, string) (string, error)
+}
+
+func (e *Encoding) SampleConfig() string {
+	return sampleConfig
+}
+
+func (e *Encoding) Description() string {
+	return "Encodes or decodes data that may also have been compressed"
+}
+
+func (e *Encoding) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	for _, point := range in {
+		valueIntf, ok := point.GetField(e.Field)
+		if !ok {
+			continue
+		}
+
+		value, ok := valueIntf.(string)
+		if !ok {
+			e.Log.Warnf("skipping encoding because metric had non-string value for field %v: %v", e.Field, valueIntf)
+			continue
+		}
+
+		var compression string
+		var compressionFound bool
+
+		if e.CompressionTag != "" {
+			compression, compressionFound = point.GetTag(e.CompressionTag)
+		}
+
+		if !compressionFound && e.CompressionField != "" {
+			if field, ok := point.GetField(e.CompressionField); ok {
+				compression, compressionFound = field.(string)
+			}
+		}
+
+		if !compressionFound {
+			compression = e.Compression
+		}
+
+		newValue, err := e.operation(value, compression, e.Encoding)
+		if err != nil {
+			e.Log.Warnf("failed to %v a metric: %v", e.Operation, err)
+			continue
+		}
+
+		point.AddField(e.DestField, newValue)
+
+		if e.RemoveOriginal {
+			point.RemoveField(e.Field)
+		}
+	}
+
+	return in
+}
+
+func (e *Encoding) Init() error {
+	switch e.Compression {
+	case "gzip":
+	case "":
+	default:
+		return fmt.Errorf("'%v' is not a supported compression algorithm. It must be 'gzip' or '' to skip.", e.Compression)
+	}
+
+	switch e.Encoding {
+	case "base64":
+	case "":
+		return fmt.Errorf("'encoding' is required for the encoding processor.")
+	default:
+		return fmt.Errorf("'%v' is not a supported encoding. It must be 'base64'.", e.Encoding)
+	}
+
+	if e.DestField == "" {
+		e.DestField = e.Field
+	}
+
+	if e.DestField == e.Field {
+		e.RemoveOriginal = false
+	}
+
+	switch e.Operation {
+	case "encode":
+		e.operation = encode
+	case "decode":
+		e.operation = decode
+	default:
+		return fmt.Errorf("'%v' is not a supported operation. It must be one of 'encode' or 'decode'.", e.Operation)
+	}
+
+	return nil
+}
+
+func newEncoding() *Encoding {
+	return &Encoding{}
+}
+
+func init() {
+	processors.Add("encoding", func() telegraf.Processor {
+		return newEncoding()
+	})
+}
+
+func encode(value, compression, encoding string) (string, error) {
+	switch compression {
+	case "gzip":
+		var buf bytes.Buffer
+		zipper := gzip.NewWriter(&buf)
+		_, err := zipper.Write([]byte(value))
+		zipper.Close()
+		if err != nil {
+			return "", err
+		}
+		value = string(buf.Bytes())
+	}
+
+	switch encoding {
+	case "base64":
+		value = base64.RawStdEncoding.EncodeToString([]byte(value))
+	}
+
+	return value, nil
+}
+
+func decode(value, compression, encoding string) (string, error) {
+	switch encoding {
+	case "base64":
+		data, err := base64.RawStdEncoding.DecodeString(value)
```

Comment:
From what I can tell this includes `/` which we talked about not using. Just wanted to make you aware in case you weren't

Reply:
Yeah, good catch. When we talked yesterday, I wasn't thinking correctly. `\` needs to be escaped; not `/`.